components/Form.js

```diff
@@ -1,15 +1,18 @@
 import { useRouter } from "next/router";
 import { styled } from "styled-components";
+import { StyledButton } from "./StyledButton";
 
-export default function Form({ submitNewCard }) {
+export default function Form({ submitNewCard, card, updateCard }) {
   const router = useRouter();
 
   function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
 
-    submitNewCard(data.name, data.icon);
+    card
+      ? updateCard(data.name, data.icon, card.id)
```

Comment:
In your user story you said you wanted to router.push the user back to the edit page after saving a new card, not the homepage. Did you change this on purpose? I personally like this option better (pushing to homepage), so it would be fine by me :)